apps/vscode-extension/src/commands/helpers.ts

```diff
@@ -26,10 +26,74 @@ export const wrapCommandWhenExplorerContext = (
   };
 };
 
+const sortWorkspaceUris = (uri1: vscode.Uri, uri2: vscode.Uri) => {
+  const pathLengthDiff = uri1.path.split(posix.delimiter).length - uri2.path.split(posix.delimiter).length;
+  if (pathLengthDiff === 0) {
+    // If path are at the same level we sort first angular.json
+    return basename(uri1.path, '.json') > basename(uri2.path, '.json')
+      ? 1
+      : -1;
+  }
+  // Else we sort first the one closer to the root level
+  return pathLengthDiff;
+};
+
+const getInfoFromWorkspaceJsonUris = async <T>(
+  getInfo: (workspaceJsonUri: vscode.Uri) => Promise<T | undefined>
+) => {
+  const workspaceJsonUris = await vscode.workspace.findFiles('{angular,nx}.json');
+  const sortedWorkspaceJsonUris = workspaceJsonUris.sort(sortWorkspaceUris);
+  const packageManagers = await Promise.all(
+    sortedWorkspaceJsonUris.map(getInfo)
+  );
+  return packageManagers.find((packageManager) => !!packageManager);
```

Comment:
You only need to read the files until you find one with the info, no need to read all of them from the start
```suggestion
  for workspaceJsonUri of sortedWorkspaceJsonUris) {
    const infoFromWorkspace = await getInfo(workspaceJsonUri);
    if (infoFromWorkspace) {
      return infoWorkspace;
    }
  }
  return {};
```